docs/api/oauth.md

```diff
@@ -0,0 +1,445 @@
+
+<br />
+
+<img src="../../static/img/oauthpopup.png" height="488" width="252" alt="Log in with Audius popup" />
+
+_Log in with Audius dialog_
+
+<br />
+
+## Workflow
+
+The "Log in with Audius" flow looks like this:
+
+1. You provide a button on your app or website to begin the authentication flow
+2. When the user clicks the button, it opens a popup containing an Audius login page that prompts the user to sign in with their Audius credentials (alternatively, your app/website can redirect to the Audius login page instead of using a popup)
+3. Once the user successfully signs in, Audius provides your app/website with the user profile using a JSON Web Token (JWT)
+4. Your app verifies and decodes the JWT
+
+The JWT payload contains the following information about the user:
+
+- Unique identifier (Audius user id)
+- Email
+- Display name
+- Audius handle
+- Whether the user is a verified artist (i.e. has a purple checkmark)
+- Profile picture URL, if any
+
+### Demo
+
+Check out a quick demo of the Oauth flow [here](https://codesandbox.io/s/log-in-with-audius-demo-j2jx6f)!
+
+### Quick links
+
+- Demo with React - [Demo app](https://j2jx6f.csb.app/) | [Code](https://codesandbox.io/s/log-in-with-audius-demo-j2jx6f?file=/src/App.js)
+- Demo with vanilla JS - [Demo app](https://xkogl3.csb.app/) | [Code](https://codesandbox.io/s/log-in-with-audius-demo-vanilla-js-xkogl3?file=/index.html)
+- [Button configurator](https://9ncjui.csb.app/)
+
+<br />
+
+## How to implement Log in with Audius with the Javascript SDK
+
+The easiest way to add Log in with Audius to your app is to use the Javascript SDK. If you are not able to use the Javascript SDK (for example, if you are developing a mobile app), skip to
+"How to implement Log in with Audius manually".
+
+Prerequisites:
+
+- [Set up and initialize the Audius SDK on your app](./sdk.md#installation)
+
+### 1. Initialize the SDK `oauth` feature
+
+First, you must call `oauth.init` and provide it with your app name, a login success callback, and an optional error callback:
+
+#### <ins>**oauth.init(loginSuccessCallback, errorCallback)**</ins>
+
+enables the Log in with Audius functionality.
+
+**Params**
+
+- loginSuccessCallback `(profile: UserProfile) => void` - function to be called when the user successfully authenticates with Audius. This function will be called with the user's profile information, which is an object with the following shape:
+
+  ```typescript
+  // type UserProfile =
+  {
+    userId: number; // unique Audius user identifier
+    email: string;
+    name: string; // user's display name
+    handle: string;
+    verified: boolean; // whether the user has the Audius "verified" checkmark
+
+    /** URLs for the user's profile picture, if any.
+     * In the vast majority of cases, three sizes will be available - 150x150, 480x480, and 1000x1000.
+    * In rare cases, only an unknown size `misc` will be available.
```

Comment:
TBH, i think we should drop the misc thing. we have plans (Q3 likely) in content team to try to fix this so that 150x150, 480x480, and 1000x1000 are always available.

would be it be too hard to just make each of these sizes point to the misc one so that we don't have to deprecate misc in the future?

Reply:
oh no not hard! i can definitely do that. so just put the one image URL in all three of the size fields?

Reply:
I would say so. Although it may be "lying" the consumer, it would functionally be fine in most cases and save them the headache down the line of us deprecating misc. wdyt?